sdk/keyvault/azure-keyvault-administration/README.md

```diff
@@ -1,29 +1,231 @@
+Once the **AZURE_CLIENT_ID**, **AZURE_CLIENT_SECRET** and
+**AZURE_TENANT_ID** environment variables are set,
+[DefaultAzureCredential][default_cred_ref] will be able to authenticate the
+[KeyVaultAccessControlClient][rbac_client_docs].
+
+Constructing the client also requires your vault's URL, which you can
+get from the Azure CLI or the Azure Portal. In the Azure Portal, this URL is
+the vault's "DNS Name".
+
+```python
+from azure.identity import DefaultAzureCredential
+from azure.keyvault.administration import KeyVaultAccessControlClient
+
+credential = DefaultAzureCredential()
+
+client = KeyVaultAccessControlClient(vault_url="https://my-key-vault.vault.azure.net/", credential=credential)
+```
 ## Key concepts
 
+### Role Definition
+A role definition defines the operations that can be performed, such as read, write, and delete. It can also define the operations that are excluded from allowed operations.
+
+A role definition is specified as part of a role assignment.
+
+### Role Assignment.
+A role assignment is the association of a role definition to a service principal. They can be created, listed, fetched individually, and deleted.
+
+### KeyVaultAccessControlClient
+A `KeyVaultAccessControlClient` provides both synchronous and asynchronous operations allowing for management of role definitions and role assignments.
+
 ## Examples
+This section conntains code snippets covering common tasks:
+* [List the role definitions](#list-the-role-definitions "List the role definitions")
+* [Create, Get, and Delete a role assignment](#create-get-and-delete-a-role-assignment "Create, Get, and Delete a role assignment")
+
+### List the role definitions
+List the role definitions available for assignment.
+
+```python
+from azure.identity import DefaultAzureCredential
+from azure.keyvault.administration import KeyVaultAccessControlClient
+
+credential = DefaultAzureCredential()
+
+client = KeyVaultAccessControlClient(vault_url="https://my-key-vault.vault.azure.net/", credential=credential)
+
+role_definitions = client.list_role_definitions(role_scope="/")  # this is the global scope
+
+for role_definition in role_definitions:
+    print(role_definition.id)
+    print(role_definition.role_name)
+    print(role_definition.description)
+```
+
+### Create, Get, and Delete a role assignment
+Assign a role to a service principal. This will require a role definition id from the list retrieved in the [above snippet](#list-the-role-definitions) and the principal object id retrieved in the [Create/Get credentials](#create/get-credentials)
+
+```python
+from azure.identity import DefaultAzureCredential
+from azure.keyvault.administration import KeyVaultAccessControlClient
+
+credential = DefaultAzureCredential()
+
+client = KeyVaultAccessControlClient(vault_url="https://my-key-vault.vault.azure.net/", credential=credential)
+
+role_scope = "/"  # setting the scope to the global scope
+role_assignment_name = "<role-assignment-name>"  # must be a UUID
```

Comment:
```suggestion
role_assignment_name = uuid.uuid4()
```
(and `import uuid` above). Note to self: would be nice to do this by default.

Reply:
Yeah that would be nice, I think .net is doing this. Another nice thing to add would be an enum for the scope like .net [here](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/keyvault/Azure.Security.KeyVault.Administration/src/KeyVaultRoleScope.cs). I can pick up both of these fixes.

Reply:
I think I was wrong about .net doing this, it just didn't show up in the readme, but it appears it shows up in the actual client @heaths for confirmation

Reply:
Yes, in .NET we do auto-create GUIDs if not passed in.